StreetHawk/Classes/Core/Private/SHDeepLinking.m

```diff
@@ -17,22 +17,108 @@
+ 6. parameter part of above, like vc=<friendly name or vc>&xib_iphone=<xib_iphone>&xib_ipad=<xib_ipad>&<additional params>
+ 
+ All above formats finally will parse to following elements:
+ 
+ * view class name: Mandatory. The name of the subclass of UIViewController, for example: @"MyViewController". If it's empty, or it's typo and cannot create the VC, return NO.
+ * iPhoneXib: Optional. The xib name for the page in iPhone, for example "MyViewController_iPhone". If it uses same name as `vcClassName`, use nil.
+ * iPadXib: Optional. The xib name for the page in iPad, for example "MyViewController_iPad". If it uses same name as `vcClassName`, use nil.
+ * dictParam: The key-value pair of deeplinking parameters, besides above elements for creating VC, it may have more custom param to pass to the page.
+ 
+ @param pushData The payload information from notification. If not used for notification pass nil.
+ @param shouldIncreaseClick Whether should try to increase Growth click_count for this deeplinking open.
+ @return If successfully create VC, return YES; otherwise, return NO.
+ */
+- (BOOL)launchDeepLinkingVC:(NSString *)deepLinking withPushData:(PushDataForApplication *)pushData increaseGrowthClick:(BOOL)shouldIncreaseClick;
+
 - (NSString *)formatXib:(NSString *)xib;  //format input string to match xib name requirement.
 - (UIViewController *)viewcontroller:(UIViewController *)vc containsSubviewcontroller:(NSString *)subVCName;  //recursive to find whether this vc contains a subview for the name.
 
 @end
 
 @implementation SHDeepLinking
 
+- (BOOL)isSupportedDeeplinkingUrl:(NSURL *)deeplinking
+{
+    NSString *command = deeplinking.host;
+    return (command != nil
+            &&( [command compare:COMMAND_LAUNCHVC options:NSCaseInsensitiveSearch] == NSOrderedSame
+                || [command compare:COMMAND_POINTZI_AUTHOR options:NSCaseInsensitiveSearch] == NSOrderedSame));
+}
+
+- (BOOL)processDeeplinkingUrl:(NSURL *)deeplinking withPushData:(PushDataForApplication *)pushData withIncreaseGrowth:(BOOL)increaseGrowth
+{
+    NSString *command = deeplinking.host;
+    if (command != nil && [command compare:COMMAND_LAUNCHVC options:NSCaseInsensitiveSearch] == NSOrderedSame) //launch vc
+    {
+        if (StreetHawk.developmentPlatform == SHDevelopmentPlatform_Native
+            || StreetHawk.developmentPlatform == SHDevelopmentPlatform_Xamarin) //automatically launch vc only available for native and xamarin
+        {
+            BOOL isHandled = [self launchDeepLinkingVC:deeplinking.absoluteString withPushData:pushData increaseGrowthClick:increaseGrowth];
+            if (isHandled)
+            {
+                return YES; //launchvc deeplinking is handled by StreetHawk
+            }
+        }
+    }
+    else if (command != nil && [command compare:COMMAND_POINTZI_AUTHOR options:NSCaseInsensitiveSearch] == NSOrderedSame) //pointzi author
```

Comment:
Please restrict line lengths ...have you configured the linter i mentiond to you